ui/src/Dashboard/actions.js

```diff
@@ -263,3 +267,259 @@ export const useActions = (flag) => {
+  }
+
+  static addCheck(check) {
+    return {
+      type: 'add_check',
+      payload: {
+        local_id: ValueOperation._localId(check.id),
+        kind: check.kind,
+        variable: check.variable,
+        operator: check.operator,
+        value_string: undefined,
+        value_number: undefined,
+        value_timestamp: undefined,
+        value_set: undefined,
+      }
+    }
+  }
+
+  static addCondition(valueId, conditionId, conditionChecks, valueConditionOverride) {
+    return {
+      type: 'add_value_condition',
+      payload: {
+        value_id: valueId,
+        local_id: ValueOperation._localId(conditionId),
+        checks: conditionChecks,
+        value_condition_override: valueConditionOverride,
+      }
+    }
+  }
+
+  static disableCondition(conditionId) {
+    return {
+      type: 'disable_value_condition',
+      payload: {
+        condition_id: conditionId
+      }
+    }
+  }
+}
+
+export function getValueSaveOperations(value, editValue, conditions, checks, variablesMap) {
+  let ops = [];
+  let errors = {};
+  let value_ = editValue;
+
+  if (!value_.dirty) {
+    return { ops, errors };
+  }
+
+  if (value.enabled === true && value_.enabled === false) {
+    ops.push(ValueOperation.disableValue(value.id));
+  } else if (value.enabled === false && value_.enabled === true) {
+    ops.push(ValueOperation.enableValue(value.id));
+  }
+
+  if (value.enabled === true && value.value_override !== value_.value_override) {
+    ops.push(ValueOperation.updateValueOverride(value.id, value_.value_override));
+  }
+
+  const originalConditions = value.conditions.map((c) => c.id);
+  const newConditions = value_.conditions;
+  forEach(difference(newConditions, originalConditions), (conditionId) => {
+    let condition = conditions[conditionId];
+    let conditionChecks = [];
+
+    forEach(condition.checks, (checkId) => {
+      if (startsWith(checkId, 'temp')) {
+        const check = checks[checkId];
+        if (check.kind === undefined) {
+          errors.missingValue = true;
+          return;
+        }
+        if (check.operator === null || check.operator === undefined) {
+          errors.missingValue = true;
+          return;
+        }
+
+        let op = ValueOperation.addCheck(check);
+        let variable = variablesMap[check.variable];
+        if (variable.type !== KIND_TO_TYPE[check.kind]) {
+          errors.internalError = true;
+          console.log(check, variable);
+          return;
+        }
+        switch (variable.type) {
+          case Type.STRING:
+            if (!isString(check.value_string) || !check.value_string) {
+              errors.internalError = true;
+              console.log(check);
+            } else {
+              op.payload.value_string = check.value_string;
+            }
+            break;
+          case Type.NUMBER:
+            if (!isNumber(check.value_number || !check.hasOwnProperty('value_number'))) {
```

Comment:
This row has a bug (introduced by me :)).

It should be `isNumber(check.value_number) ||` not `isNumber(check.value_number ||`

Reply:
And the same bug exists in getFlagSaveOperations